crates/extensions/tedge_mqtt_bridge/src/lib.rs

```diff
@@ -351,19 +506,12 @@ async fn half_bridge(
             Event::Incoming(Incoming::Publish(publish)) => {
                 if let Some(publish) = loop_breaker.ensure_not_looped(publish).await {
                     if let Some(topic) = transformer.convert_topic(&publish.topic) {
-                        target
-                            .publish(
-                                topic.clone(),
-                                publish.qos,
-                                publish.retain,
-                                publish.payload.clone(),
-                            )
-                            .await
-                            .unwrap();
-                        companion_bridge_half
-                            .send(Some((topic.into_owned(), publish)))
-                            .await
-                            .unwrap();
+                        received += 1;
+                        target.publish(topic.to_string(), publish).await;
+                    } else {
+                        // Being not forwarded to this bridge target
+                        // The message has to be acknowledged
+                        recv_client.ack(&publish).await.unwrap()
```

Comment:
Good point. This shouldn't ever be the case, but it is the correct way to handle it. We should be able to convert any incoming message as the subscription is based on the mapping rules.